src/util/dataJoin.js

```diff
@@ -46,7 +46,7 @@
             // enter
             // when container is a transition, entering elements fade in (from transparent to opaque)
             var enterSelection = updateSelection.enter()
-                .append(element) // <<<--- this is the secret sauce of this whole file
+                .insert(element) // <<<--- this is the secret sauce of this whole file (#528)
```

Comment:
I'd explicitly say we're using the special enter-only insert rather than append to help with ordering